### ISSUE TITLE
Use srcset on for <amp-img>

### ISSUE THREAD

**Issue by kristoferbaxter** [ampproject/amp.dev#3244]
# Is your feature request related to a problem? Please describe.
Images loaded on `amp.dev` do not use `srcset` to choose a smaller image for different screen sizes. Using srcset would reduce the number of bytes sent for image requests.

```html
<amp-img class="ap--media i-amphtml-layout-responsive i-amphtml-layout-size-defined" src="/static/img/case-band-image-2.jpg.webp" alt="A screenshot of a website using AMP." layout="responsive" width="519" height="931" i-amphtml-layout="responsive"><i-amphtml-sizer style="display:block;padding-top:179.3834%;"></i-amphtml-sizer>
  <amp-img fallback="" class="ap--media i-amphtml-layout-responsive i-amphtml-layout-size-defined" src="/static/img/case-band-image-2.jpg" alt="A screenshot of a website using AMP." layout="responsive" width="519" height="931" i-amphtml-layout="responsive">
    <i-amphtml-sizer style="display:block;padding-top:179.3834%;"></i-amphtml-sizer>
  </amp-img>
</amp-img>
```

Here using src set could reduce the size of these images significantly for smaller resolution devices.

# Describe the solution you'd like
Leverage srcset for the default `webp` image and the fallback `jpg`.

# Optional: Teachability, Documentation, Adoption, Migration Strategy
This relatively small change would have a significant impact on the number of bytes sent on the homepage of `amp.dev` and would provide material for a blog post.

**Comment by DerekNonGeneric** [comment]
I would like to take care of this. Can someone please assign the issue to me?

**Comment by CrystalOnScript** [comment]
Thanks @DerekNonGeneric - assigned!

**Comment by sebastianbenz** [comment]
@DerekNonGeneric thanks for offering to look into this! 

Few comments about this:

* I'd like to avoid solving this in our build system. The problem is: our amp.dev build image is already 1GB big. Having images in multiple dimensions would further increase this.
* My preferred way to solve this would be:
   1. adding an [AMP Optimizer transformer](https://github.com/ampproject/amp-toolbox/tree/master/packages/optimizer/lib/transformers) that automatically generates secret tags if they don't exist yet.
   2. use either an existing Google Cloud image hosting solution or setup [Thumbor](https://github.com/thumbor/thumbor) on GCP.

These constraints probably make this non as straightforward to solve as it first seemed. I'm happy to support you if you're still willing to tackle this.

**Comment by DerekNonGeneric** [comment]
@sebastianbenz, sure thing man! I would like to share with you an idea I had about how to set this up during dev mode without incurring the optimization cost: create symlinks for all the srcset images. I haven't tested it, but it should work! 

Prior research:

- the srcset code should probably go in the [`image.j2` macro](https://github.com/ampproject/amp.dev/blob/future/frontend/templates/macros/image.j2)
- [Responsive Images](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images) reference

Feel free to reassign, looking forward to your solution!

**Comment by patrickkettner** [comment]
fixed in #3978
